### PR TITLE
krb5 1.20.1 and add dependencies

### PR DESCRIPTION
**Pull request by DE0CH**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Building `krb5` requires `libedit` for the header `editline/readline.h`.`krb5` builds fine on debian for some reason but it didn't build on other linux distros that don't have the header file somewhere. In any case, including `libedit` should be safer.

### CODE REVIEW

File: Formula/krb5.rb [hunk 1]
```diff
@@ -25,6 +25,10 @@ class Krb5 < Formula
 
   uses_from_macos "bison"
 
+  on_linux do
+    depends_on "libedit" => :build
+  end
```
Comment (carlocab) [inline Formula/krb5.rb:30]:
```suggestion
  uses_from_macos "libedit" => :build
  uses_from_macos "bison"
```

**Review by carlocab — COMMENTED** [review]
Please use the preferred commit-message style for homebrew/core. We put the name of the formula first in commit-message headings.

For new formulae:

> At Homebrew, we like to put the name of the formula up front like so: `foobar 7.3 (new formula)`.

For existing formulae:

> The preferred commit message format for simple version updates is `foobar 7.3` and for fixes is `foobar: fix flibble matrix.`.

Refer to the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit) for more details. Also, when making further changes to your pull request, use the following guidelines to make sure that @BrewTestBot can merge your commits:

> - One formula per commit; one commit per formula.
> - Keep merge commits out of the pull request.

**Comment by DE0CH** [comment]
Sorry about the commit style issue. I have fixed the problems. Please review.

**Comment by Bo98** [comment]
> Building `krb5` requires `libedit` for the header `editline/readline.h`.

This should only happen if `pkg-config` is detecting libedit is installed:

https://github.com/krb5/krb5/blob/krb5-1.20-final/src/configure.ac#L1340-L1376
https://github.com/krb5/krb5/blob/krb5-1.20-final/src/util/ss/listen.c#L17-L18

Though it's probably a good idea to depend on libedit or readline anyway. Though I don't think it's build-only.

**Comment by DE0CH** [comment]
That's interesting. On my system (which is running Scientific Linux 7.9 with a ton of really old packages), it failed with the make step (not the configure setup). 

> In any case, I don't think it's build-only.

I see, I'll remove the build only option

**Comment by Bo98** [comment]
Alternative would be to just pass `--without-libedit` and `--without-readline` if the benefits are limited. I don't particularly mind though.

**Comment by DE0CH** [comment]
I would prefer having libedit because it is quite lite and it can prevent having problems with other packages that depend on it.

**Comment by DE0CH** [comment]
Not sure why CI too so long to finish that it got canceled.

**Comment by carlocab** [comment]
It takes a long time because this formula has many dependents.

**Comment by carlocab** [comment]
I'll review this once we've had the chance to give this a full CI run.

**Comment by DE0CH** [comment]
Can you re-trigger CI? I don't think I can trigger it (besides pushing).

**Comment by DE0CH** [comment]
CI failed because of the following error
```
No CI-syntax-only label found. Running tests job.


No CI-no-fail-fast label found. Stopping tests on first failing matrix build.

No CI-skip-dependents label found. Running brew test-bot --only-formulae-dependents.

No CI-long-timeout label found. Setting short GitHub Actions timeout.

```
Error: PR requires the CI-long-timeout label but it is not set!
Error: If the longer timeout is not required, remove the "long build" label.
Error: Otherwise, add the "CI-long-timeout" label.
Error: No more than 2 PRs at a time may use "CI-long-timeout".
No CI-test-bot-fail-fast label found. Not passing --fail-fast to brew test-bot.
No CI-build-dependents-from-source label found. Not passing --build-dependents-from-source to brew test-bot.
No CI-skip-recursive-dependents label found. Not passing --skip-recursive-dependents to brew test-bot.
No CI-skip-livecheck label found. Not passing --skip-livecheck to brew test-bot.
```

**Comment by SMillerDev** [comment]
Yes, CI needs a special slot to schedule a CI run for this and such a slot is not available right now.

**Comment by cho-m** [comment]
The common way of packaging `krb5` on Linux would be to use `libss.so` from `e2fsprogs` (as done by Alpine, Arch, Debian, etc.). I think the `krb5` internal copy is where the `libedit` dependency comes from.

However, in Homebrew, it may be simpler to just use `libedit` for now or disable it.

If we were to properly package `e2fsprogs` (current formula is a bit off compared to common distribution), then Linux would require depending on `util-linux` as that formula has main libuuid/blkid and that would pull in a couple other dependencies: `libxcrypt`, `ncurses`, `zlib`.

Since Homebrew doesn't supporting split/sub packages which can reduce dependency trees (in this case, we shouldn't need libuuid/etc but cannot avoid those), pulling in `e2fsprogs` may not be ideal as `krb5` is a heavily used dependency on Linux.

---

As side note, if we were to pick one, `libedit` is probably better than `readline` in this case to avoid propagating GPL licensing restrictions. Would only impact the corresponding library (`libss.so`), but makes it more likely to hit license conflicts. 

Though, in other cases, `readline` may be better as it is the GNU variant which is more commonly used on Linux.

**Comment by DE0CH** [comment]
I don't really know much about the package dependencies in linux, so I am quite lost with your comment. 

Are you saying using `libedit` in the current form is ok? Or is there anything that need to be changed in the formula?

**Comment by SMillerDev** [comment]
Re-running the build. @carlocab and @cho-m any other comments here?

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**